test/support/sycl_alloc_utils.h

```diff
@@ -13,8 +13,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef __TEST_SYCL_ALLOC_UTILS_H
-#define __TEST_SYCL_ALLOC_UTILS_H
+#ifndef _TEST_SYCL_ALLOC_UTILS_H
+#define _TEST_SYCL_ALLOC_UTILS_H
```

Comment:
I'm not sure if this is something we want to tackle with this PR, but the naming convention for test header guards is not consistent.  
I think we need to decide the scope of this PR.  If we are interested in making the test header guards fully consistent within this PR, then there is work to do.  If not, then these changes are still beneficial, and I don't have a problem having them tag along with the rest of the changes.

Reply:
If we are OK with not including the test files in the scope of this PR for full consistency, then I think the rest of my issues are resolved.

Reply:
Fixed.